src/other/tables.jl

```diff
@@ -55,21 +55,18 @@ Tables.columns(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.columns(pare
 Tables.rows(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.rows(parent(itr))
 Tables.schema(itr::Union{DataFrameRows,DataFrameColumns}) = Tables.schema(parent(itr))
 Tables.materializer(itr::DataFrameRows) =
-    eachrow ∘ prefer_singleton_callable(Tables.materializer(parent(itr)))
+    x -> eachrow(Tables.materializer(parent(itr))(x))
 function Tables.materializer(itr::DataFrameColumns)
-    f = prefer_singleton_callable(Tables.materializer(parent(itr)))
```

Comment:
Why not just have two methods for `DataFrameColumns` instead of one method with `eltype` check? But this is a mild preference (maybe wait for @nalimilan to comment before changing this 😄)

Reply:
Yes that would sound more logical.